sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/joins.scala

```diff
@@ -194,12 +194,12 @@ object EliminateOuterJoin extends Rule[LogicalPlan] with PredicateHelper {
     }
   }
 
-  private def allDuplicateAgnostic(
-      aggregateExpressions: Seq[NamedExpression]): Boolean = {
-    !aggregateExpressions.exists(_.exists {
-      case agg: AggregateFunction => !EliminateDistinct.isDuplicateAgnostic(agg)
-      case _ => false
-    })
+  private def allDuplicateAgnostic(a: Aggregate): Boolean = {
+    a.groupOnly || a.aggregateExpressions.flatMap { e =>
+      e.collect {
+        case ae: AggregateExpression => ae
+      }
+    }.forall(ae => ae.isDistinct || EliminateDistinct.isDuplicateAgnostic(ae.aggregateFunction))
```

Comment:
This optimization make sense. Shall we also take `a.distinctKeys` ? i.e., `a.child.distinctKeys.contains(ae.child)`